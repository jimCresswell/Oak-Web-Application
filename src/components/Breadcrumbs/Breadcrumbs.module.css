.nav {
  overflow: hidden;
  margin-bottom: 24px;
}

.ol {
  display: flex;
  align-items: center;
  flex-wrap: wrap;
  list-style: none;
  margin: 0;
  padding: 0;
}

.li {
  display: flex;
  align-items: center;
  font-weight: bold;
  font-size: 16px;
  margin: 0;
  padding: 0;
  white-space: nowrap;
  max-width: 100%;
}

.a {
  overflow: hidden;
  text-overflow: ellipsis;
  flex: 1;
  min-width: 0;
  max-width: 100%;
}
