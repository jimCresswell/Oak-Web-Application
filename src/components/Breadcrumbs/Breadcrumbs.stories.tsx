import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import Component from "./Breadcrumbs";

export default {
  title: "Navigation/Breadcrumbs",
  component: Component,
  argTypes: {},
} as ComponentMeta<typeof Component>;

const Template: ComponentStory<typeof Component> = (args) => (
  <Component {...args} />
);

export const Breadcrumbs = Template.bind({});
Breadcrumbs.args = {
  breadcrumbs: [
    { href: "/", label: "Unit Quiz" },
    { href: "/", label: "View In Classroom" },
    { href: "/", label: "Foundation Curriculum (PDF)" },
    { href: "/", label: "Higher Curriculum (PDF)" },
  ],
};
