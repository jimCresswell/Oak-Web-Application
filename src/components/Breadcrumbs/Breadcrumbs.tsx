import { FC } from "react";
import Link from "next/link";
import styled from "styled-components";

import { BreadcrumbJsonLd } from "../../browser-lib/seo/getJsonLd";
import Icon from "../Icon";
import UL from "../Typography/UL";
import ellipsis from "../../styles/ellipsis";
import flex from "../../styles/utils/flex";

const BreadcrumbsNav = styled.nav`
  display: flex;
  min-width: 0;
`;

const BreadcrumbUL = styled(UL)`
  ${flex};
`;

const BreadcrumbsLi = styled.li`
  display: flex;
  align-items: center;
  min-width: 0;
  flex-shrink: 0;

  &:last-of-type {
    flex-shrink: 2;
  }
`;

const BreadcrumbConstrainer = styled.div`
  ${ellipsis}
`;

export type Breadcrumb = {
  href: string;
  label: string;
  disabled?: boolean;
};
export type BreadcrumbsProps = {
  breadcrumbs: Breadcrumb[];
};
const Breadcrumbs: FC<BreadcrumbsProps> = ({ breadcrumbs }) => {
  return (
    <>
      <BreadcrumbJsonLd itemListElements={breadcrumbs} />
      <BreadcrumbsNav aria-label="Breadcrumb">
        <BreadcrumbUL $reset $minWidth={0} $flexWrap={"nowrap"}>
          {breadcrumbs.map((breadcrumb, i) => {
            const { href, label, disabled } = breadcrumb;
            return (
              <BreadcrumbsLi key={`${i}-${href}`}>
                {i !== 0 && (
                  <Icon
                    name="ChevronRight"
                    size={20}
                    $color={"teachersHighlight"}
                  />
                )}
                <BreadcrumbConstrainer>
                  {disabled ? <>{label}</> : <Link href={href}>{label}</Link>}
                </BreadcrumbConstrainer>
              </BreadcrumbsLi>
            );
          })}
        </BreadcrumbUL>
      </BreadcrumbsNav>
    </>
  );
};

export default Breadcrumbs;
