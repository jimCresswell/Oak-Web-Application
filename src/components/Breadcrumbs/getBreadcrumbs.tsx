import { BlogWebinarCategory } from "../../common-lib/cms-types";
import { SerializedWebinar } from "../../pages/beta/webinars/[webinarSlug]";
import { SerializedBlog } from "../../pages/blog/[blogSlug]";
import { Breadcrumb } from "../Breadcrumbs";

/*
 *
 * getBlogBreadcrumbs()
 *
 * takes a list of blog categories and the slug of the selected category
 * returns an array of breadcrumbs - blog > [category]
 *
 */

export const getBlogWebinarListBreadcrumbs = (
  categories: BlogWebinarCategory[],
  currentCategorySlug: string | null,
  page: "blog" | "beta/webinars", //@TODO: remove "beta/"
  label: string
): Breadcrumb[] => [
  { label, href: `/${page}` },
  {
    label:
      categories.find((cat) => cat.slug === currentCategorySlug)?.title ||
      "All",
    href: currentCategorySlug || `/${page}`,
    disabled: true,
  },
];

export const getBlogWebinarPostBreadcrumbs = (
  categories: BlogWebinarCategory[],
  blog: SerializedBlog | SerializedWebinar,
  page: "blog" | "beta/webinars", //@TODO: remove "beta/"
  label: string
): Breadcrumb[] => {
  const { title, slug, category } = blog;
  return [
    { label, href: `/${page}` },
    {
      label:
        categories.find((cat) => cat.slug === category.slug)?.title || "All",
      href: `/${page}/categories/${category.slug}`,
    },
    {
      label: title,
      href: slug,
      disabled: true,
    },
  ];
};
