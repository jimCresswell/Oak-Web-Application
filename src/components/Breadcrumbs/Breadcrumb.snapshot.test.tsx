import { render } from "@testing-library/react";

import Breadcrumbs from "./Breadcrumbs";

const breadcrumbs = [
  { href: "/", label: "Unit Quiz" },
  { href: "/", label: "View In Classroom" },
  { href: "/", label: "Foundation Curriculum (PDF)" },
  { href: "/", label: "Higher Curriculum (PDF)" },
];

it("renders <Breadcrumb> unchanged", () => {
  const { container } = render(<Breadcrumbs breadcrumbs={breadcrumbs} />);
  expect(container).toMatchSnapshot();
});
