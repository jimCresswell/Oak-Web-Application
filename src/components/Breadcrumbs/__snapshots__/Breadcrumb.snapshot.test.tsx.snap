// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Breadcrumb> unchanged 1`] = `
<div>
  <nav
    aria-label="Breadcrumb"
    class="nav"
  >
    <ol
      class="ol"
    >
      <li
        class="li"
      >
        <a
          href="/"
        >
          Unit Quiz
        </a>
      </li>
      <li
        class="li"
      >
        <span
          class="outer-wrapper"
          style="height: 20px; width: 20px;"
        >
          <span
            class="inner-wrapper"
            style="height: 20px; width: 20px;"
          >
            <svg
              fill="none"
              height="100%"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2.5"
              viewBox="0 0 24 24"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="9 18 15 12 9 6"
              />
            </svg>
          </span>
        </span>
        <a
          href="/"
        >
          View In Classroom
        </a>
      </li>
      <li
        class="li"
      >
        <span
          class="outer-wrapper"
          style="height: 20px; width: 20px;"
        >
          <span
            class="inner-wrapper"
            style="height: 20px; width: 20px;"
          >
            <svg
              fill="none"
              height="100%"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2.5"
              viewBox="0 0 24 24"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="9 18 15 12 9 6"
              />
            </svg>
          </span>
        </span>
        <a
          href="/"
        >
          Foundation Curriculum (PDF)
        </a>
      </li>
      <li
        class="li"
      >
        <span
          class="outer-wrapper"
          style="height: 20px; width: 20px;"
        >
          <span
            class="inner-wrapper"
            style="height: 20px; width: 20px;"
          >
            <svg
              fill="none"
              height="100%"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2.5"
              viewBox="0 0 24 24"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="9 18 15 12 9 6"
              />
            </svg>
          </span>
        </span>
        <a
          href="/"
        >
          Higher Curriculum (PDF)
        </a>
      </li>
    </ol>
  </nav>
</div>
`;
